library/core/src/ptr/alignment.rs

```diff
@@ -131,6 +167,91 @@ impl fmt::Debug for Alignment {
     }
 }
 
+#[unstable(feature = "ptr_alignment_type", issue = "102070")]
+impl fmt::Display for Alignment {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        fmt::Display::fmt(&self.as_nonzero(), f)
+    }
+}
+
+#[unstable(feature = "ptr_alignment_type", issue = "102070")]
+impl fmt::Binary for Alignment {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        fmt::Binary::fmt(&self.as_nonzero(), f)
+    }
+}
+
+#[unstable(feature = "ptr_alignment_type", issue = "102070")]
+impl fmt::Octal for Alignment {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        fmt::Octal::fmt(&self.as_nonzero(), f)
+    }
+}
+
+#[unstable(feature = "ptr_alignment_type", issue = "102070")]
+impl fmt::LowerHex for Alignment {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        fmt::LowerHex::fmt(&self.as_nonzero(), f)
+    }
+}
+
+#[unstable(feature = "ptr_alignment_type", issue = "102070")]
+impl fmt::UpperHex for Alignment {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        fmt::UpperHex::fmt(&self.as_nonzero(), f)
+    }
+}
+
+impl Alignment {
+    // longest string: 1p-128
```

Comment:
even though no targets have 128-bit usize, I figured I might as well future-proof it